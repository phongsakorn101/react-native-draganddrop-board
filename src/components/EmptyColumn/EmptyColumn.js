import React from 'react'
import {
  number,
  string
} from 'prop-types'
import {
  colors,
  fonts
} from '../../constants'
import { Empty } from '../Icons'
import {
  EmptyWrapper,
  Paragraph
} from './EmptyColumn.styled'

const EmptyColumn = ({
  emptyIconColor,
  emptyText,
  emptyTextColor,
  emptyTextFontFamily,
  emptyTextFontSize,
  marginTop
}) =>  (
  <EmptyWrapper marginTop={marginTop}>
    <Empty color={emptyIconColor} />
    <Paragraph
      color={emptyTextColor}
      fontFamily={emptyTextFontFamily}
      fontSize={emptyTextFontSize}
    >
      {emptyText}
    </Paragraph>
  </EmptyWrapper>
)

EmptyColumn.defaultProps = {
  emptyIconColor: colors.blurple,
  emptyText: 'No Tasks',
  emptyTextColor: colors.bay,
  emptyTextFontFamily: '',
  emptyTextFontSize: 16
}

EmptyColumn.propTypes = {
  emptyIconColor: string.isRequired,
  emptyText: string.isRequired,
  emptyTextColor: string.isRequired,
  emptyTextFontFamily: string.isRequired,
  emptyTextFontSize: number.isRequired,
  marginTop: number.isRequired
}

export default EmptyColumn
