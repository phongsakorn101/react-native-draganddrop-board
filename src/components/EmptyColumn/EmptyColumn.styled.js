import styled from 'styled-components/native'
import {
  color,
  fontFamily,
  fontSize
} from 'styled-system'

const EmptyWrapper = styled.View`
  justifyContent: center;
  alignItems: center;
`

const Paragraph = styled.Text`
  ${fontFamily};
  ${fontSize};
  ${color};
`

export {
  EmptyWrapper,
  Paragraph
}
