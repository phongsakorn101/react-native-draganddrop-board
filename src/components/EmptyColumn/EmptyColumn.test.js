import React from 'react'
import renderer from 'react-test-renderer'
import EmptyColumn from './EmptyColumn'

describe('EmptyColumn component', () => {
  describe('Renders correctly', () => {
    test('it renders Default EmptyColumn', () => {
      const tree = renderer.create(
        <EmptyColumn marginTop={40} />
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
