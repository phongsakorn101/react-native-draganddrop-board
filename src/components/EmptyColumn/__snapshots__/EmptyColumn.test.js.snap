// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyColumn component Renders correctly it renders Default EmptyColumn 1`] = `
<View
  marginTop={40}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/icons/emptyBox.png",
      }
    }
    style={
      Object {
        "height": 40,
        "tintColor": "#4834d4",
        "width": 40,
      }
    }
  />
  <Text
    color="#747d8c"
    fontFamily=""
    fontSize={16}
    style={
      Array [
        Object {
          "color": "#747d8c",
          "fontSize": 16,
        },
      ]
    }
  >
    No items
  </Text>
</View>
`;
